### PR TITLE
Preserve old version of releases manifest for backwards compatibility

### PR DESCRIPTION
**Pull request by abhay-krishna**
Partially reverting #4777 to have both the older and newer formats of the releases manifest so as to maintain backwards compatibility when performing latest minor release upgrades.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/api/v1alpha1/release_types.go [hunk 1]
```diff
@@ -87,9 +87,23 @@ type EksARelease struct {
 	// +kubebuilder:validation:Required
 	// EKS Anywhere binary bundle
 	EksABinary BinaryBundle `json:"eksABinary"`
```
Comment (abhay-krishna) [inline release/api/v1alpha1/release_types.go:89]: This field will be removed in later releases.

**Review by jonathanmeier5 — COMMENTED** [review]
Just one question, otherwise lgtm.

File: release/pkg/test/testdata/release-0.14-bundle-release.yaml [hunk 2]
```diff
@@ -1920,7 +1920,7 @@ spec:
       version: v0.0.0-dev-release-0.14+build.0+abcdef1
     etcdadmBootstrap:
       components:
-        uri: https://release-bucket/artifacts/v0.0.0-dev-release-0.14-build.0/etcdadm-bootstrap-provider/manifests/bootstrap-etcdadm-bootstrap/v1.0.5-rc3/bootstrap-components.yaml
+        uri: https://release-bucket/artifacts/v0.0.0-dev-release-0.14-build.0/etcdadm-bootstrap-provider/manifests/bootstrap-etcdadm-bootstrap/v1.0.5-rc4/bootstrap-components.yaml
```
[the remark this answers is not in the document]
Comment (abhay-krishna) [reply release/pkg/test/testdata/release-0.14-bundle-release.yaml:1923]: this is just a testdata file, it matches the tag values in the build tooling repo

**Pull request merged**